tests/src/Interop/PInvoke/IEnumerator/IEnumeratorNative.h

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <xplatform.h>
+#include <platformdefines.h>
+#include <algorithm>
+
+class IntegerEnumerator : public IEnumVARIANT
+{
+    ULONG refCount = 1;
+    int start;
+    int count;
+    int current;
+
+public:
+    IntegerEnumerator(int start, int count)
+        :start(start),
+        count(count),
+        current(start)
+    {
+    }
+
+    HRESULT STDMETHODCALLTYPE Next( 
+        ULONG celt,
+        VARIANT *rgVar,
+        ULONG *pCeltFetched) override
+    {
+        for(*pCeltFetched = 0; *pCeltFetched < celt && current < start + count; ++*pCeltFetched, ++current)
+        {
+            VariantClear(&(rgVar[*pCeltFetched]));
+            V_VT(&rgVar[*pCeltFetched]) = VT_I4;
+            V_I4(&(rgVar[*pCeltFetched])) = current;
+        }
+        
+        return celt == *pCeltFetched ? S_OK : S_FALSE;
+    }
+
+    HRESULT STDMETHODCALLTYPE Skip(ULONG celt) override
+    {
+        int original = current;
+        current = std::min(current + (int)celt, start + count);
+        return original + (int)celt <= start + count ? S_OK : S_FALSE;
+    }
+
+    HRESULT STDMETHODCALLTYPE Reset(void) override
+    {
+        current = start;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE Clone(IEnumVARIANT **ppEnum) override
+    {
+        IntegerEnumerator* clone = new(CoreClrAlloc(sizeof(IntegerEnumerator))) IntegerEnumerator(start, count);
+        clone->current = current;
+        *ppEnum = clone;
+
+        return S_OK;
+    }
+
+    ULONG STDMETHODCALLTYPE AddRef() override
+    {
+        return ++refCount;
+    }
+
+    ULONG STDMETHODCALLTYPE Release() override
+    {
+        ULONG newRefCount = --refCount;
+
+        if (newRefCount == 0)
+        {
+            CoreClrFree(this);
+        }
+
+        return newRefCount;
+    }
+
+    HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void** ppIntf)
+    {
+        HRESULT hr = E_NOINTERFACE;
+        if (iid == __uuidof(IUnknown))
+        {
+            *ppIntf = dynamic_cast<IUnknown*>(this);
```

Comment:
Bah. Please don't ever use `dynamic_cast<>` - especially from the Windows compiler. Use `static_cast<>` instead.